samcli/lib/warnings/sam_cli_warning.py

```diff
@@ -0,0 +1,85 @@
+"""
+Provides all Warnings checkers for sam template
+"""
+import logging
+
+LOG = logging.getLogger(__name__)
+
+
+class TemplateWarning:
+    """
+    Top level class which all warnings should extend from.
+    """
+
+    def check(self, template_dict):
+        raise Exception("NotImplementedException")
+
+
+def _get_deployment_preferences_status(function):
+    """
+    Takes a AWS::Serverless::Function resource and checks if resource have a deployment preferences applied
+    to it. If DeploymentPreference found then it returns its status if it is enabled or not.
+    """
+    deployment_preference = function["Properties"].get("DeploymentPreference", None)
```

Comment:
should we do function.get("Properties", {}).get("DeploymentPreference", None) just to avoid failures.